src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioSuppressionFixService.cs

```diff
@@ -110,7 +143,7 @@ public bool RemoveSuppressions(bool selectedErrorListEntriesOnly)
                 }
 
                 var project = solution.GetProject(diagnosticsByProject.Key);
-                if (project != null)
+                if (project != null && shouldFixInProject(project))
```

Comment:
i feel like this lambda isn't needed at all.  You can just have an instance function you call, passing in the project to it.